server/test/com/xsn/explorer/util/RetryableFutureSpec.scala

```diff
@@ -0,0 +1,140 @@
+package com.xsn.explorer.util
+
+import org.scalatest.{AsyncWordSpec, MustMatchers}
+import scala.concurrent.Future
+import akka.actor.{ActorSystem, Scheduler}
+import scala.concurrent.duration._
+import scala.util.Try
+
+class RetryableFutureSpec extends AsyncWordSpec with MustMatchers {
+
+  class UnstableOperation(numberOfFailures: Int, error: Throwable) {
+    var tryCount: Int = 0
+    var lastExecutionTime: Long = 0
+    private var delays: List[FiniteDuration] = List.empty[FiniteDuration]
+
+    def getDelays = {
+      delays.reverse
+    }
+
+    def execute: Future[Int] = {
+      if (lastExecutionTime > 0) {
+        delays = (System.currentTimeMillis - lastExecutionTime).millis :: delays
+      }
+      lastExecutionTime = System.currentTimeMillis
+
+      Future {
+        if (tryCount == numberOfFailures) {
+          1
+        } else {
+          tryCount = tryCount + 1
+          throw error
+        }
+      }
+    }
+  }
+
+  object UnstableOperation {
+
+    def apply(numberOfFailures: Int, error: Throwable): UnstableOperation = {
+      new UnstableOperation(numberOfFailures, error)
+    }
+  }
+
+  implicit val scheduler: Scheduler = ActorSystem().scheduler
```

Comment:
You need to make sure the actor system is terminated after the test completes, for example:
```
class RetryableFutureSpec
    extends TestKit(ActorSystem("RetryableFutureSpec"))
    with WordSpecLike
    with BeforeAndAfterAll {

  override def afterAll: Unit = {
    TestKit.shutdownActorSystem(system)
  }
}
```